src/main/default/classes/TriggerConfigService.cls

```diff
@@ -0,0 +1,115 @@
+public class TriggerConfigService {
+    public static ObjectConfig getConfig(String objName) {
+        String lowercaseObjName = objName.toLowerCase();
+        if (!configMetadata.keySet().contains(lowercaseObjName)) {
+            return null;
+        }
+
+        return configMetadata.get(lowercaseObjName);
+    }
+
+    public static Map<String, ObjectConfig> configMetadata {
+        get {
+            if (configMetadata == null) {
+                configMetadata = buildConfigMap();
+            }
+            return configMetadata;
+        }
+        set;
+    }
+
+    // @testvisible
+    // private static void overrideHandlerEnabled(String objectName, String className) {
+    // }
+    @testVisible
+    private static Map<String, ObjectConfig> buildConfigMap() {
+        Trigger_Object__mdt[] allMtd = [
+            SELECT
+                Object__c,
+                Enabled__c,
+                (
+                    SELECT
+                        Handler_Class__c,
+                        Auto_Bind__c,
+                        Enabled__c,
+                        Execution_Order__c,
+                        Before_Insert__c,
+                        After_Insert__c,
+                        Before_Delete__c,
+                        After_Delete__c,
+                        After_Undelete__c,
+                        Before_Update__c,
+                        After_Update__c
+                    FROM Trigger_Handlers__r
+                    ORDER BY Execution_Order__c ASC
+                )
+            FROM Trigger_Object__mdt
+        ];
+        Map<String, ObjectConfig> resultsMap = new Map<String, ObjectConfig>();
+        for (Trigger_Object__mdt objMdt : allMtd) {
+            resultsMap.put(objMdt.Object__c, new ObjectConfig(objMdt));
+        }
+        return resultsMap;
+    }
+
+    public class ObjectConfig {
+        public String objName;
+        public Boolean enabled;
+        public Map<String, HandlerConfig> handlers;
+        public String[] handlerOrder;
+        public Set<String> disabledHandlers;
+
+        public ObjectConfig() {
+            handlers = new Map<String, HandlerConfig>{};
+            disabledHandlers = new Set<String>();
+            handlerOrder = new List<String>{};
```

Comment:
nitpick, can initialize in the declaration ... then you don't need to repeat this code in both constructors